tools/gengen/util/gengen.go

```diff
@@ -271,42 +261,43 @@ func setupMiners(vm consensus.GenesisVM, st state.Tree, keys []*types.KeyInfo, m
 		}
 
 		// give collateral to account actor
-		_, err = vm.ApplyGenesisMessage(vmaddr.LegacyNetworkAddress, addr, types.SendMethodID, abi.NewTokenAmount(100000), nil)
+		_, err = vm.ApplyGenesisMessage(vmaddr.LegacyNetworkAddress, addr, builtin.MethodSend, abi.NewTokenAmount(100000), nil)
 		if err != nil {
 			return nil, err
 		}
 
-		ret, err := vm.ApplyGenesisMessage(addr, vmaddr.StoragePowerAddress, power.CreateStorageMiner, abi.NewTokenAmount(100000), power.CreateStorageMinerParams{
-			OwnerAddr:  addr,
-			WorkerAddr: addr,
-			PeerID:     pid,
-			SectorSize: types.NewBytesAmount(m.SectorSize),
+		// Note: the owner of the miner is implicit in the msg. The owner will be the `from` address.
+		out, err := vm.ApplyGenesisMessage(addr, vmaddr.StoragePowerAddress, builtin.MethodsPower.CreateMiner, abi.NewTokenAmount(100000), &power.CreateMinerParams{
+			Worker:     addr,
+			Peer:       pid,
+			SectorSize: abi.SectorSize(m.SectorSize),
 		})
 		if err != nil {
 			return nil, err
 		}
 
 		// get miner ID address
-		mIDAddr := ret.(address.Address)
+		ret := out.(*power.CreateMinerReturn)
 
 		minfos = append(minfos, RenderedMinerInfo{
-			Address: mIDAddr,
+			Address: ret.IDAddress,
 			Owner:   m.Owner,
 			Power:   types.NewBytesAmount(m.SectorSize * m.NumCommittedSectors),
 		})
 
-		// add power directly to power table
-		for i := uint64(0); i < m.NumCommittedSectors; i++ {
-			powerReport := types.NewPowerReport(m.SectorSize*m.NumCommittedSectors, 0)
-
-			_, err := vm.ApplyGenesisMessage(addr, vmaddr.StoragePowerAddress, power.ProcessPowerReport, abi.NewTokenAmount(0), power.ProcessPowerReportParams{
-				Report:     powerReport,
-				UpdateAddr: mIDAddr,
-			})
-			if err != nil {
-				return nil, err
-			}
-		}
+		// Dragons: we need a new way of doing this
```

Comment:
fyi @ZenGround0 I think this is where you come in (but probably by messages for pre/committing sectors)

Reply:
100% same page